src/controller/CHIPDeviceController.cpp

```diff
@@ -1206,6 +1206,8 @@ CHIP_ERROR DeviceCommissioner::SendOperationalCertificateSigningRequestCommand(C
     Callback::Cancelable * successCallback = mOpCSRResponseCallback.Cancel();
     Callback::Cancelable * failureCallback = mOnCSRFailureCallback.Cancel();
 
+    MutableByteSpan csrNonce = device->GetCSRNonce();
+    ReturnErrorOnFailure(mOperationalCredentialsDelegate->GenerateNOCSR(csrNonce));
```

Comment:
Please use a temporary buffer to get hold the nonce locally, and then use `CHIP_ERROR SetCSRNonce(ByteSpan csrNonce)`. It is cleaner than trying to gain access to the underlying memory of the DeviceProxy

Reply:
Addressed in latest commit